platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/AjaxEngine.ts

```diff
@@ -0,0 +1,166 @@
+// class definition of DOMContext.
+class AjaxEngine {
+	private readonly nodes = new Map<number, Node>();
+	private zIndex = 100;
+
+	public allocateZIndex() {
```

Comment:
In general, we could explicitly define return types for functions. Should we?
That way, we could avoid unintended (and probably unnoticed) changes to method return types (which are currently inferred).

Reply:
I don't think that would improve the code, but rather increase unnecessary clutter....if you change the implicit return type and that would make calling code invalid, we will notice it.

Reply:
Hm. Fair point.